audbackend/core/backend/minio.py

```diff
@@ -177,7 +177,8 @@ def _checksum(
     ) -> str:
         r"""MD5 checksum of file on backend."""
         path = self.path(path)
-        return self._client.stat_object(self.repository, path).etag
+        meta = self._client.stat_object(self.repository, path).metadata
+        return meta["x-amz-meta-checksum"] if "x-amz-meta-checksum" in meta else ""
```

Comment:
I am  thinking about the meaning of `if "x-amz-meta-checksum" in meta else` . 
What are the conditions in which `metadata` does not contain the `x-amz-meta-checksum` field? 

I see that it is retrieved by `self._client.stat_object`, and this can be calculated on local files (as in ` self._checksum(src_path)` in the `_copy_file` method. 

So my assumption would have been that it is always there, right?

Reply:
Yes, `x-amz-meta-` is always there, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html#UserMetadata

Reply:
But why then use `if "x-amz-meta-checksum" in meta else ""`?

Reply:
I see.  I have a problem overseeing what could be the worst case that happens by a missing (empty) checksum.

Reply:
This is what the AI says when running it locally: 

```
suggestion(edge_case_not_handled): Consider handling missing checksum more explicitly.
Returning an empty string when the checksum is missing might lead to silent failures or unexpected behavior. Consider raising an exception or logging a warning to make it clear that the checksum is not available.

```

Reply:
Thanks for raising this concern, this is indeed worth some additonal thought.

The worst case that can happen is that files will be transferred again (which also happens in the current `main` branch as the wrong MD5 sum is returned for larger files). E.g. if `put_file()` finds the same filename on the server it checks if the checksum match, and if not upload the file again. The checksum match is calculated as a string comparison, so it will not result in an error (which would be the case if we return `None` instead of an empty string).

I would not raise an exception or a warning, as `audbackend` should also work for servers, that contain files, that were uploaded with different software. In that case, they will most likely not have a `x-amz-meta-checksum` entry.
But maybe we should change the names to `x-amz-meta-audbackend-checksum` and `x-amz-meta-audbackend-owner` to avoid colliding with metadata created by other software?

One alternative I see for returning `""` if the metadata entry is missing, is returning the `etag`, which we do currently in `main`. Any preference what to return?

Reply:
>> One alternative I see for returning "" if the metadata entry is missing, is returning the etag, which we do currently in main. Any preference what to return?

This merge request deliberately departs from `etag` as it can become wrong for large files. I would think that the current checksum is an improvement in that sense.

Reply:
Yes I would say so too. The values are there to validate data artifact identity, and the worst case again is copying back and fro data. Therefore resolving.